homestar-runtime/src/cli/init.rs

```diff
@@ -0,0 +1,366 @@
+        .write_all(settings_toml.as_bytes())
+        .expect("to write settings file");
+
+    Ok(())
+}
+
+fn handle_quiet(quiet: bool) -> Result<Box<dyn Write>> {
+    if quiet {
+        Ok(Box::new(empty()))
+    } else {
+        Ok(Box::new(stdout()))
+    }
+}
+
+fn handle_output_mode(
+    output_mode: OutputMode,
+    no_input: bool,
+    force: bool,
+    writer: &mut Box<dyn Write>,
+) -> Result<Box<dyn Write>> {
+    match output_mode {
+        OutputMode::StdOut => Ok(Box::new(stdout())),
+        OutputMode::File { path } if force => {
+            if let Some(parent) = path.parent() {
+                std::fs::create_dir_all(parent).expect("to create parent directory");
+            }
+
+            let settings_file = File::options()
+                .read(true)
+                .write(true)
+                .create(true)
+                .truncate(true)
+                .open(&path)
+                .expect("to open settings file");
+
+            writeln!(writer, "Writing settings to {:?}", path).expect("to write");
+
+            Ok(Box::new(settings_file))
+        }
+        OutputMode::File { path } => {
+            if let Some(parent) = path.parent() {
+                std::fs::create_dir_all(parent).expect("to create parent directory");
+            }
+
+            let settings_file = File::options()
+                .read(true)
+                .write(true)
+                .create_new(true)
+                .open(&path);
+
+            // This seemingly convoluted match is to avoid the risk of a
+            // TOCTOU race condition, where another process creates the file
+            // in between this one checking for its existence and opening it.
+            let settings_file = match settings_file {
+                Ok(file) => file,
+                Err(err) if err.kind() == std::io::ErrorKind::AlreadyExists => {
+                    if no_input {
+                        bail!("Aborting... settings file already exists at {:?}. Pass `--force` to overwrite it", path);
+                    }
+
+                    let should_overwrite = Confirm::new(&format!(
+                        "Settings file already exists at {:?}, overwrite?",
+                        path
+                    ))
+                    .with_default(false)
+                    .prompt()
+                    .map_err(|e| miette!(e))?;
+
+                    if !should_overwrite {
+                        bail!("Aborting... not overwriting existing settings file");
+                    }
+
+                    File::options()
+                        .read(true)
+                        .write(true)
+                        .create_new(false)
+                        .open(&path)
+                        .expect("to open settings file")
+                }
+                err => err.expect("to open settings file"),
+            };
+
+            writeln!(writer, "Writing settings to {:?}", path).expect("to write");
```

Comment:
b/c we could also be writing out the PEM (whether defaulted or given), can we have a note to the console saying that? The case I realized:

didn't overwrite settings, but pem was written in the previous step without any ack.